app/src/router.ts

```diff
@@ -21,13 +22,18 @@ export const defaultRoutes: RouteRecordRaw[] = [
 	{
 		name: 'login',
 		path: '/login',
-		component: LoginRoute,
-		props: (route) => ({
-			ssoErrorCode: route.query.error ? route.query.code : null,
-			logoutReason: route.query.reason,
-		}),
+		components: {
+			default: LoginRoute,
+			notice: LoginRouteNotice,
+		},
+		props: {
+			notice: (route) => ({
+				logoutReason: route.query.reason,
+			}),
```

Comment:
Added benefit of named-router-view based slotting: individual prop definition and extraction for each component